airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -161,8 +164,7 @@ public void decrement(final long stateId, final long count) {
    *
    * @return list of state messages with no more inflight records.
    */
-  public List<PartialStateWithDestinationStats> flushStates() {
-    final List<PartialStateWithDestinationStats> output = new ArrayList<>();
+  public void flushStates() {
```

Comment:
suggestion: what if `flushStates` took in an output record collector? This helps us prevent passing the record collector everywhere. Should help comprehension.